packages/nns/src/token.ts

```diff
@@ -0,0 +1,100 @@
+import { ICPTs } from "../proto/ledger_pb";
+import { E8S_PER_TOKEN, TOKEN_DECIMAL_ACCURACY } from "./constants/constants";
+import { FromStringToTokenError } from "./enums/token.enums";
+
+export const convertStringToE8s = (
+  amount: string
+): bigint | FromStringToTokenError => {
+  // Remove all instances of "," and "'".
+  amount = amount.trim().replace(/[,']/g, "");
+
+  // Verify that the string is of the format 1234.5678
+  const regexMatch = amount.match(/\d*(\.\d*)?/);
+  if (!regexMatch || regexMatch[0] !== amount) {
+    return FromStringToTokenError.InvalidFormat;
+  }
+
+  const [integral, fractional] = amount.split(".");
+
+  let e8s = BigInt(0);
+
+  if (integral) {
+    try {
+      e8s += BigInt(integral) * E8S_PER_TOKEN;
+    } catch {
+      return FromStringToTokenError.InvalidFormat;
+    }
+  }
+
+  if (fractional) {
+    if (fractional.length > 8) {
+      return FromStringToTokenError.FractionalMoreThan8Decimals;
+    }
+    try {
+      e8s += BigInt(fractional.padEnd(8, "0"));
+    } catch {
+      return FromStringToTokenError.InvalidFormat;
+    }
+  }
+
+  return e8s;
+};
+
+export type Token = {
+  symbol: string;
+  name: string;
+};
+
+export const ICPToken: Token = {
+  symbol: "ICP",
+  name: "ICP",
+};
+
+export class TokenAmount {
+  protected constructor(protected e8s: bigint, public token: Token) {}
+
+  public static fromE8s(amount: bigint, token: Token = ICPToken): TokenAmount {
```

Comment:
can we use object as param?

for this and following function

`{amount, token}: {amount: bigint, token: Token}`

Reply:
ok

Reply:
done